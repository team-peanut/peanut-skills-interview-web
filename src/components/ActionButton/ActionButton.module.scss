@use "../../styles/typography";

.wrapper {
  @include typography.typography-button;

  background: var(--action-color);
  border-radius: 100px;
  border: 0;
  color: white;
  padding: calc(var(--spacing-small) / 2) var(--spacing-regular);
  width: min-content;
}
