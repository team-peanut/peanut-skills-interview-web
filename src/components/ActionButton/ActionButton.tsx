import React from 'react'
import classNames from "classnames";


import styles from './ActionButton.module.scss';

export interface Props {
    children?: React.ReactNode
    className?: string
}

function ActionButton({children, className}: Props): JSX.Element {

    return (
        <button type="button" className={classNames(styles.wrapper, className)}>{children}</button>
    );
};

export default ActionButton;
