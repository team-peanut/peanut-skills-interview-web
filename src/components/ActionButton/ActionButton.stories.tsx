import React from "react";
import { ComponentStory } from "@storybook/react";

import ActionButton, {Props} from "./ActionButton";

export default {
  title: "Components/ActionButton",
  component: ActionButton,
};

const Template: ComponentStory<typeof ActionButton> = (props: Props) => (
  <ActionButton  />
);

export const Primary = Template.bind({});
Primary.args = {
  
};



