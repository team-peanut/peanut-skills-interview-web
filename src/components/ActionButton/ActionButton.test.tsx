import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";


import ActionButton, {Props} from "./ActionButton";


describe("ActionButton", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  
  afterEach(cleanup);

  it("renders as expected", () => {
    const { asFragment } = render(<ActionButton><>Hello World</></ActionButton>);
    expect(asFragment()).toMatchSnapshot();
  });

});
