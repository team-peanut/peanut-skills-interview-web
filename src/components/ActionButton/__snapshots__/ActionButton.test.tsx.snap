// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButton renders as expected 1`] = `
<DocumentFragment>
  <button
    class="wrapper"
    type="button"
  >
    Hello World
  </button>
</DocumentFragment>
`;
